### PR TITLE
Initial Workshop Docs

### PR DESCRIPTION
**Pull request by Equalizer5118**
Simple workshop docs, with  a quick what is it page and steps on how to upload an addon. Images included. Maybe with proper links? 

Please suggest additions or changes, its probably not going to be right on the first try!

### CODE REVIEW

File: docs/modding/workshop/workshopgui.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+title: Workshop Uploader GUI
+---
+
```
Comment (JJL772) [inline docs/modding/workshop/workshopgui.md:4]: Can you add something like this here:
```
# Workshop Uploader

The workshop uploader is a GUI frontend that can upload addons to the Steam workshop.
Addons may contain maps, models, textures, materials and anything else that can be put in a normal Source Mod.
```

Just needs to be a simple summary.

**Review by JJL772 — CHANGES_REQUESTED** [review]
Looks good, couple different comments tho!!

File: docs/modding/workshop/workshopoverview.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: "Portal 2: Community Edition Workshop"
+---
+
+# The Workshop: What Is It?
+Like Portal 2, Portal 2: Community Edition has a workshop where users can upload their own maps for others to play. Portal 2: CE's workshop, however, can do much more than just maps. Similar to the Left 4 Dead 2 and Black Mesa workshops, entire mod-packs or multi-map campaigns can be in the same addon. But it doesn't have to be just maps, addons can consist of just material reskins or remodels, or even entire game ports. 
+
+# How do I upload an addon to the workshop?
```
[the remark this answers is not in the document]
Comment (Equalizer5118) [reply docs/modding/workshop/workshopoverview.md]: Fixed in recent commit

File: docs/modding/workshop/workshopgui.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# Setting up your Addon
+Before uploading your addon, you must make sure all of your addon's assets (maps, materials, scripts, models, etc) are all in the right folders. Your Addon should be structured similar to a custom folder or sourcemod, with the root folder being your MOD folder and the subfolders being where your assets are located. It is usually the best practice to put your addon inside of `p2ce\addons`, as the game can register your addon as a local addon and you can manage it like any other addon from the workshop inside of the game. Your folder setup should look similar to this:
+
+![Addon folder](images/addon_pre.png)
+
+Now, we can proceed with making this an actual addon.
+
+# How to use the Workshop Uploader
```
[the remark this answers is not in the document]
Comment (Equalizer5118) [reply docs/modding/workshop/workshopgui.md]: ditto as above

File: docs/modding/workshop/workshopgui.md (hunk 1 above)
Comment (Equalizer5118) [reply docs/modding/workshop/workshopgui.md:4]: also added in recent commit

File: docs/modding/workshop/workshopoverview.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+title: "Portal 2: Community Edition Workshop"
```
Comment (JoLoZ) [inline docs/modding/workshop/workshopoverview.md:2]: Please use game neutral titles, if features (eg. the workshop) are only supported in some games, use [feature flags](https://wiki.stratasource.org/contribute/basics/feature-flags).

Comment (Equalizer5118) [reply docs/modding/workshop/workshopoverview.md:2]: Alright, how would I go about adding a new feature flag? The workshop this documentation is referring to is for P2CE only, but the wiki doesn't have a - WORKSHOP feature flag.

Comment (JJL772) [reply docs/modding/workshop/workshopoverview.md:2]: Yeah, I'd just call this page "Workshop" instead.

Comment (JoLoZ) [reply docs/modding/workshop/workshopoverview.md:2]: Just normalized the feature flags, the one for workshop is `USE_UGC`.

Comment (Equalizer5118) [reply docs/modding/workshop/workshopoverview.md:2]: Alright, done
Ive also added - USE_UGC feature flag to both pages

**Review by JJL772 — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**